spec/Vehicle/DigitalKey.vspec

```diff
@@ -0,0 +1,236 @@
+#
+# Key State (Primary and secondary keys)
+#
+# Each key (primary key or secondary keys) has an associated state.
+# The state indicates for example if the key is correctly setup and valid.
+#
+# For more information see DigitalKey.KeyData.KeyState
+#
+# There are several reasons why a key state may change. Note that the implementation is outside the scope of this spec.
+#
+# Examples (informative):
+# - If the primary key paring state changes to UNPAIRED all existing secondary keys may be terminated or removed.
+# - If the vehicle identity (DigitalKey.VehicleId) changes all existing keys may be terminated or removed.
+#
+#
+# Key Capabilities and Key Presence
+#
+# Keys may have different capabilities. Some keys can be used to drive the vehicle while others
+# only can be used for specific purposes or with specific restrictions.
+# The VSS model does not make any assumptions on what capabilities exist, but includes a signal
+# DigitalKey.KeyData.KeyCapability (of string datatype) which can be used to show capabilities
+# of the current key.
+#
+# Multiple digital keys may be in the vicinity of the vehicle simultaneously.
+# The VSS model include two signals (DigitalKey.KeyPresence and DigitalKey.KeyPresenceInVehicle)
+# that show which keys that are present in the vicinity of the vehicle and inside the vehicle, respectively.
+
+DigitalKey:
+  type: branch
+  description: Data related to management of Digital Keys
+
+DigitalKey.VehicleId:
+  datatype: uint8[]
+  type: attribute
+  description: Identifier used to identify the vehicle for digital key purposes.
+  comment: The vehicle identifier used for key purposes is not necessarily the same as any identifier used
+           for other purposes. For privacy reasons, it is likely not the same as the VIN.
+           The identifier may change during the lifetime of the vehicle, for example at owner change
+           to effectively invalidate all existing digital keys.
+
+DigitalKey.PairingState:
+  datatype: string
+  type: sensor
+  allowed: ['UNPAIRED', 'PAIRED', 'PAIRING','BLOCKED']
+  description: Pairing State for the primary digital key.
+               UNPAIRED indicates that no valid primary key exists for the vehicle.
+               PAIRED indicates that pairing has been performed and valid primary key has been registered.
+               PAIRING indicates that pairing is in progress.
+               BLOCKED indicates that pairing is disabled.
+
+DigitalKey.WirelessCapabilities:
+  type: attribute
+  datatype: string
+  allowed: ['NFC', 'NFC_BLE', 'NFC_BLE_UWB']
+  description: Specifies which wireless capabilities the digital key system in the vehicle supports.
+               NFC indicates NFC Forum Analog and Digital Technical Specification 2.1.
+               BLE indicates Bluetooth Core Specification 5.0 or later.
+               UWB indicates IEEE Standard for Low-Rate Wireless Networks (802.15.4z-2020).
+
+DigitalKey.BluetoothLeAddress:
```

Comment:
Presumably, this attribute is only present or set for a BLE DigitalKey, correct? What would be returned if this attribute is not set? All zeroes? Should a comment mention this?

Similar comment for other fields in this definition.

Reply:
Historically we typically do not include any "undefined" value for VSS signals. So that if you ask for `Vehicle.VehicleIdentification.VehicleExteriorColor` or `Vehicle.Speed` and your VSS server currently does not know the vehicle color or the vehicle speed, the API should return "not available", "not found", "None" or whatever mechanism the API has to indicate that no value exist.

(This could off course be discussed for this particular case; if we would benefit from have a specific value for undefined/unknown)